runners/core-java/src/test/java/org/apache/beam/runners/core/SimpleDoFnRunnerTest.java

```diff
@@ -386,6 +388,89 @@ public void testInfiniteSkew() {
             BoundedWindow.TIMESTAMP_MAX_VALUE));
   }
 
+  /**
+   * Demonstrates that attempting to set a timer with an output timestamp before the timestamp of
+   * the current element with zero {@link DoFn#getAllowedTimestampSkew() allowed timestamp skew}
+   * throws.
+   */
+  @Test
+  public void testTimerBackwardsInTimeNoSkew() {
```

Comment:
It would make more sense to make these tests `@ValidatesRunner` tests then use the deprecated `DoFnRunner`. This will allow us to get coverage across multiple implementations.

Reply:
I added ValidatesRunner tests for both the timer case and the normal DoFn case as additional tests. Unit tests still make sense for this specific DoFn and the validates runner ones cannot check for the exact error without being overly restrictive. This seems like the wrong place for tests that validate a runner but it seems like that's standard practice in beam so I just put them here.